server/client.go

```diff
@@ -3937,6 +3941,13 @@ func (c *client) processInboundClientMsg(msg []byte) (bool, bool) {
 	}
 	c.mu.Unlock()
 
+	// Check if the client is trying to publish to reserved NRG subjects.
+	// Doesn't apply to NRGs themselves as they use SYSTEM-kind clients instead.
+	if c.kind == CLIENT && bytes.HasPrefix(c.pa.subject, clientNRGPrefix) && acc != c.srv.SystemAccount() {
```

Comment:
Still have some things here we are accessing from c.

Maybe add a new lock for looking up the system account?

Reply:
Client lock was released just above, so there is no lock ordering issue. If the worry is about accessing `c.kind` and `c.srv`, these could be captured above under the lock. `c.pa.subject` should not be an issue since this is "thread local" to the processing of an inbound message.

If the concern is accessing server lock here just to see if the account is system account, maybe adding an atomic boolean in the `Account` struct that is set when setting the system account and could be checked here with atomic opeartion?

Reply:
`c.kind` doesn't change after the creation of the client connection, `c.pa.subject` is indeed thread-local and `c.srv` also doesn't change. I think this is safe.

Note there are existing accesses to `c.kind` and `c.pa.subject` further up outside of the client lock.

Reply:
Still think we may want separate server mutex around looking up system account. Or make it an atomic.Pointer.

Reply:
Or just compare against s.sys with no locks.